### PR TITLE
automate patch-level release

### PR DESCRIPTION
**Pull request by namusyaka**
Hello @sinatra friends!
I've been working on the automation recently, and right now I confirmed the automation works correctly, so I would like to get your feedback!

Background: #1510 

This change offers two features.
1. Automate to upgrade `CHANGELOG.md` and `VERSION` for next patch level release.
2. Execute `rake release:all` to publish all gems.

We're now able to cut a patch-level release easily by using github labels:

### `release/patch`
- If the pull request has this label and gets merged, new patch version will be released. (e.g. `v2.0.6`)

### `release/candidate`
- If the pull request has this label and gets merged, new version will be released as release-candidate. (e.g. `v2.0.6.rc`)

### `release-note`
- The automation script will automatically generate the CHANGELOG entry.
- The script will collect pull requests for generating entry by whether the label has been attached or not.
---
Before merging this, we need to check some points for reducing troubles.

We need to check that required access has been granted for @sinatra-bot.
@sinatra-bot should be:
- Added as a `collaborator` (read/write) into this repository
- Added an owner to `sinatra`, `sinatra-contrib` and `rack-protection` gems in rubygems.org.

Then, I have a private key for @sinatra-bot, and it must be encrypted and be committed to this repository for accessing to this repository from travisci, so we need encrypt the private key by `travis encrypt-file [PEM]`. You can see more details in .travis.yml and release.sh.

---
Last, thanks for taking a look at this pull request. If you have more time, please review the codes, the structure, and actual automation workflow at https://github.com/sinatra-bot/sandbox.
I'm looking forward to get your feedback!

Thanks,

### CODE REVIEW

File: Rakefile [hunk 1]
```diff
@@ -200,6 +200,17 @@ if defined?(Gem)
       end
     end
 
+    desc "Prepares for the patch release"
+    task :travis do
+      load "ci/release.rb"
+      sh <<-SH
+        echo "---\n:rubygems_api_key: #{ENV['RUBYGEMS_API_KEY']}" > ~/.gem/credentials
+        cat ~/.gem/credentials
```
Comment (olleolleolle) [inline Rakefile]: Does this need output here?

Comment (namusyaka) [reply Rakefile]: I totally forgot about this, but seems a debug code to me

Comment (namusyaka) [reply Rakefile]: I'll try to take a look in a few days.

Comment (olleolleolle) [reply Rakefile:208]: I just went over this again, and noted our conversation here. Good to see your avatar again. Hope things are good! 👋

**Comment by namusyaka** [comment]
Closing this as we should use Github Actions now.

**Pull request closed without merging**